src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -159,7 +159,17 @@ grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
 
   // Step 2. Create the first file chunk for this file and allocate chunk
   // servers
-  return HandleFileChunkCreation(request, reply);
+  grpc::Status chunk_creation_status(HandleFileChunkCreation(request, reply));
+
+  // If we did not create chunk successfully during file creation, we roll back
+  // and remove the file metadata and chunk metadata that got created along
+  // the way.  
+  if (!chunk_creation_status.ok()) {
+    LOG(ERROR) << "Rolling back and deleting file metadata: " << filename;
+    metadata_manager()->DeleteFileMetadata(filename);
```

Comment:
this deletes the file metadata, but it doesn't delete the chunk handle created, if the failure was after creating the chunk handle (such as initializing file chunks)

Reply:
Forgot to reply this one. Actually this deletes all chunk handles and all associated chunk metadata. I think the name is a misnomer, and I have renamed it to DeleteFileAndChunkMetadata.